packages/react-ai/src/components/AIConversation/views/Controls/FieldControl.tsx

```diff
@@ -84,31 +91,72 @@ const TextInput: typeof TextAreaBase = React.forwardRef(
       <TextAreaBase
         {...props}
         data-testid="text-input"
-        placeholder={isFirstMessage ? 'Ask anything...' : 'Message Raven'}
+        onChange={(e) =>
+          props.onChange ??
+          (setInput &&
+            setInput((prevInput) => ({ ...prevInput, text: e.target.value })))
+        }
+        placeholder={
+          props.placeholder ?? isFirstMessage
+            ? 'Ask anything...'
+            : 'Message Raven'
+        }
         ref={ref}
       />
     );
   }
 );
 
-const Container = withBaseElementProps(View, {
-  className: `${FIELD_BLOCK}__container`,
+const InputContainer = withBaseElementProps(View, {
+  className: `${FIELD_BLOCK}__input-container`,
 });
 
 export const FieldControl: FieldControl = () => {
+  const { input, setInput } = React.useContext(InputContext);
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    (e.target as HTMLFormElement).reset();
+
+    if (input?.text) {
```

Comment:
I assume we might want a util that does this but not sure where it would belong

Reply:
Hm yeah we can add that once we get the sendMessage api hooked up